git_pull/config.yaml

```diff
@@ -28,7 +27,7 @@ options:
     - .gitignore
   git_command: pull
   git_prune: false
-  deployment_key: []
+  deployment_key_base64_encoded: ""
```

Comment:
> Also, I decided to pass the SSH deployment key base64 encoded; otherwise, parsing the single-line SSH key would be a lot of struggle.

Uh I see. Well but this is weird too, asking the user to base64 encode their private keys yet again. The PEM (`-----BEGIN...` `-----END`...) format is already base64 encoded.

Now granted, how the multiple lines are handled here is a bit funky, even in YAML terms. E.g. instead of:

```yaml
deployment_key:
  - "-----BEGIN RSA PRIVATE KEY-----"
  - MIIEowIBAAKCAQEAv3hUrCvqGZKpXQ5ofxTOuH6pYSOZDsCqPqmaGBdUzBFgauQM
  - xDEcoODGHIsWd7t9meAFqUtKXndeiKjfP0MMKsttnDohL1kb9mRvHre4VUqMsT5F
  - "..."
  - i3RUtnIHxGi1NqknIY56Hwa3id2yk7cEzvQGAAko/t6PCbe20AfmSQczs7wDNtBD
  - HgXRyIqIXHYk2+5w+N2eunURIBqCI9uWYK/r81TMR6V84R+XhtvM
  - "-----END RSA PRIVATE KEY-----"
```

We could use [multiline YAML](https://yaml-multiline.info/) (note the pipe after the colon):

```yaml
deployment_key: |
  -----BEGIN RSA PRIVATE KEY-----
  MIIEowIBAAKCAQEAv3hUrCvqGZKpXQ5ofxTOuH6pYSOZDsCqPqmaGBdUzBFgauQM
  xDEcoODGHIsWd7t9meAFqUtKXndeiKjfP0MMKsttnDohL1kb9mRvHre4VUqMsT5F
  ...
  i3RUtnIHxGi1NqknIY56Hwa3id2yk7cEzvQGAAko/t6PCbe20AfmSQczs7wDNtBD
  HgXRyIqIXHYk2+5w+N2eunURIBqCI9uWYK/r81TMR6V84R+XhtvM
  -----END RSA PRIVATE KEY-----
```

But I guess you want to address the UI representation. I am not aware that we can request multi-line string from the frontend.

So we have a bit unfortunate limitations from the UI side currently :cry: 

However, what we could do as a work around is using sub-option. Those are displayed as YAML fields. How about getting rid of `deployment_key` and `deployment_key_protocol`, and move to:

```
schema:
  deployment_identity:
    type: match(rsa|dsa|ecdsa|ed25519)
    key: str
```

Adjust the docs to show an example how to do the multiline string thing:

![image](https://github.com/home-assistant/addons/assets/34061/241fb5b6-fcab-4e78-b474-e081a1ea7911)

:thinking: 


In any case. This would need a major number bump as it breaks backwards compatibility (and set the `breaking_versions` config).